core/vm/evm.go

```diff
@@ -314,8 +320,10 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 			precompiles = PrecompiledContractsStaking
 			writeCapablePrecompiles = WriteCapablePrecompiledContractsStaking
 		}
-
-		if writeCapablePrecompiles[addr] == nil && precompiles[addr] == nil && evm.ChainConfig().IsS3(evm.EpochNumber) && value.Sign() == 0 {
+		if evm.chainRules.IsCrossShardXferPrecompile {
+			writeCapablePrecompiles = WriteCapablePrecompiledContractsCrossXfer
+		}
+		if (len(writeCapablePrecompiles) == 0 || writeCapablePrecompiles[addr] == nil) && precompiles[addr] == nil && evm.ChainConfig().IsS3(evm.EpochNumber) && value.Sign() == 0 {
```

Comment:
why add a len() check in the condition?

Reply:
The code prior to this change was initializing `writeCapablePrecompiles` with `make`. Now it is no longer initialized, so this check is needed to avoid a `nil` error.
https://github.com/harmony-one/harmony/blob/ec8d197f00aa51ce47c6403609373d2bef14f7e1/core/vm/evm.go#L306